### PR TITLE
Add better YML parsing for metadata

### PR DESCRIPTION
**Pull request by cloudnautique**
This is to address #2298

### CODE REVIEW

File: rancher/util.go [hunk 1]
```diff
@@ -1,5 +1,70 @@
 package rancher
 
+import (
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+)
+
+func InterfaceToString(str interface{}) string {
+	switch str.(type) {
```
Comment (ibuildthecloud) [inline rancher/util.go]: could you just do `fmt.Sprintf("%v", str)`?  would do that same thing?

File: rancher/util.go [hunk 2]
```diff
@@ -1,5 +1,70 @@
...
+
+func NestedMapsToMapInterface(data map[string]interface{}) map[string]interface{} {
+	newMapInterface := map[string]interface{}{}
+
+	for k, v := range data {
+		switch v.(type) {
+		case map[interface{}]interface{}:
+			v = mapWalk(v.(map[interface{}]interface{}))
+		case []interface{}:
+			if listHasMaps(v.([]interface{})) {
```
Comment (ibuildthecloud) [inline rancher/util.go]: what if its a list of a list of maps.  Do you really have to check if it has a map?  Just blindy convert.

File: rancher/util.go (hunk 1 above)
Comment (cloudnautique) [inline rancher/util.go]: Yes, Im going to remove it. Less code to deal with. thanks.

File: rancher/util.go (hunk 2 above)
Comment (cloudnautique) [inline rancher/util.go]: Talked it over with @cjellick going to push the traversal into mapWalk and rename mapWalk.

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/2298

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/2301

**Comment by ibuildthecloud** [comment]
This still doesn't cover list of a list of maps

**Pull request merged**